### PR TITLE
"Queuecaster" feature enhancement

### PR DESCRIPTION
**Pull request by RobertK66**
This adds following Cast Messages/Models:  QueueLoad, QueueNext, QueuePrev, QueueChange, QueueItemIds, QueueItems

What I did: 
- updated all referenced packages to lates stable
- added the "parallelAlgorithm": "aggressive" option into XUnit runnsettings.
   ( without that the 'sequenced' Media Channel Tests do fail since XUNIT version 2.8.0 introduced different default value )
- made all UnitTests to run in sequence by adding all into one "SingleCollection"
- refactored and added MediaChannelTests
- removed some code in MediaChannel.cs avoiding duplicated MediaStatus events.
- added the features mentioned above
- Wrote an 'awkward manual' Unittest to proof the included fix for #226 
- added constructor option to provide ILogger/ILoggerFactory without having to provide the own set of services.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,3 +4,5 @@ updates:
     directory: "/" # Location of package manifests
     schedule:
       interval: "daily"
+    ignore:
```
Comment (Tapanila) [inline .github/dependabot.yml:7]: can you remove the ignore part?

File: Sharpcaster/Sharpcaster.csproj [hunk 2]
```diff
@@ -1,17 +1,17 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <PackageId>Sharpcaster</PackageId>
     <RepositoryType>git</RepositoryType>
-    <RepositoryUrl>https://github.com/Tapanila/SharpCaster/</RepositoryUrl>
+    <RepositoryUrl>https://github.com/RobertK66/SharpCaster/</RepositoryUrl>
```
Comment (Tapanila) [inline Sharpcaster/Sharpcaster.csproj:7]: can you change this back?

File: Sharpcaster/Models/Media/Item.cs [hunk 3]
```diff
@@ -24,6 +24,16 @@ public class Item
...
+        public bool Autoplay { get; set; } = true;
+
+
+        [DataMember(Name = "orderId")]
+        public long OrderId { get; set; }
+
+        [DataMember(Name = "startTime")]
+        public long StartTime { get; set; }
+
+        //[DataMember(Name = "preloadTime")]
```
Comment (Tapanila) [inline Sharpcaster/Models/Media/Item.cs:36]: can you add these here:
        [DataMember(Name = "preloadTime")]
        public long PreloadTime { get; set; }

        [DataMember(Name = "activeTrackIds")]
        public long[] ActiveTrackIds { get; set; }

**Comment by Tapanila** [comment]
I made some changes on https://github.com/Tapanila/SharpCaster/commit/43788e4530cc9973e09453ad2efe8a5d1e05945d.
If you can just pull those changes in I'm fine with merging

**Comment by RobertK66** [comment]
Wow, thx. for cleaning up my messy left behinds and non formatting ;-)....

What I think you missed, is that I intentionally used and expanded your existing Media.Item class because this is the same object used in the MediaStatus answer and in the QueueLoad (see [here](https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.MediaStatus#items) and [here](https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.QueueLoadRequestData#items)  ). So I think we should delete the Media.Item class and always use the QueueItem for that. You can also see in the LOG after a QueueLoad the mediaStatus contains 2 items (current and next) and both of them have the orderId field set! 
The separation in the 2 Namespaces is therefore not that clean as you tried to do it 😄 ....

BTW: Does the "CD" list of audio tracks somehow show on a Cast Video device (unfortunately I do not have one) !?

**Comment by Tapanila** [comment]
I think it makes more sense to remove the Media.Item class because it's pretty empty class currently.

My SharpCasterDemoApplication that's used on the testing doesn't have a UI elements to show what's playing so there's no feedback on what it's playing.

**Comment by RobertK66** [comment]
But can you hear the Audio?

**Comment by Tapanila** [comment]
Yep and all controls work fine

**Pull request merged**